src/docker/build.nim

```diff
@@ -269,10 +269,25 @@ proc launchDockerSubscan(ctx:     DockerInvocation,
   result = runChalkSubScan(usableContexts, "insert").report
   trace("docker: subscan complete.")
 
+proc getRelativePathBetween(base_path: string, path: string) : string =
+  ## Given the `base_path`, usually the project root, return the relative
+  ## path of the file's `path`. Return nothing if its outside the project root
+  ## or if `path` is an empty string.
+  let relativePath = path.relativePath(base_path)
+  if relativePath.startsWith("..") or relativePath == "":
+    trace("Dockerfile is ephemeral or not contained within VCS project")
```

Comment:
strictly this util is not just for the dockerfile

Reply:
should i pop it over to `src/util.nim`?